packages/mdc-checkbox/mdc-checkbox.scss

```diff
@@ -14,333 +14,135 @@
-    @include mdc-theme-prop(border-color, primary);
-    @include mdc-theme-prop(background-color, primary);
-
-    transition:
-      mdc-checkbox-transition-enter(border-color),
-      mdc-checkbox-transition-enter(background-color);
-
-    .mdc-checkbox__checkmark {
-      transition:
-        mdc-checkbox-transition-enter(opacity, 0ms, $mdc-checkbox-transition-duration * 2),
-        mdc-checkbox-transition-enter(transform, 0ms, $mdc-checkbox-transition-duration * 2);
-      opacity: 1;
-
-      &__path {
-        stroke-dashoffset: 0;
-      }
-    }
 
-    .mdc-checkbox__mixedmark {
-      transform: scaleX(1) rotate(-45deg);
+  @include mdc-theme-dark {
+    @include mdc-checkbox-background-unfilled-enabled {
+      @include mdc-checkbox-stroke-color($mdc-checkbox-border-color-dark);
     }
-  }
-}
-
-.mdc-checkbox__native-control:indeterminate {
-  ~ .mdc-checkbox__background {
-    @include mdc-theme-prop(border-color, primary);
-    @include mdc-theme-prop(background-color, primary);
 
-    .mdc-checkbox__checkmark {
-      transform: rotate(45deg);
-      transition:
-        mdc-checkbox-transition-exit(opacity, 0ms, $mdc-checkbox-transition-duration),
-        mdc-checkbox-transition-exit(transform, 0ms, $mdc-checkbox-transition-duration);
-      opacity: 0;
-
-      &__path {
-        stroke-dashoffset: 0;
-      }
+    @include mdc-checkbox-background-unfilled-disabled {
+      @include mdc-checkbox-stroke-color($mdc-checkbox-disabled-color-dark);
     }
 
-    .mdc-checkbox__mixedmark {
-      transform: scaleX(1) rotate(0deg);
-      opacity: 1;
+    @include mdc-checkbox-background-filled-disabled {
+      @include mdc-checkbox-container-color($mdc-checkbox-disabled-color-dark);
     }
   }
-}
-
-.mdc-checkbox__native-control:disabled,
-[aria-disabled="true"] .mdc-checkbox__native-control {
-  cursor: default;
-
-  // postcss-bem-linter: ignore
-  ~ .mdc-checkbox__background {
-    border-color: $mdc-checkbox-disabled-color;
 
-    @include mdc-theme-dark(".mdc-checkbox") {
-      border-color: $mdc-checkbox-disabled-color-dark;
-    }
-  }
+  @include mdc-checkbox-focus-color(secondary);
+  @include mdc-checkbox-state-transition-color(secondary);
+  @include mdc-checkbox-ripple-color(secondary);
 
-  &:checked,
-  &:indeterminate {
-    ~ .mdc-checkbox__background {
-      border-color: transparent;
-      background-color: $mdc-checkbox-disabled-color;
-      // stylelint-disable selector-max-compound-selectors, selector-max-specificity
-      @include mdc-theme-dark(".mdc-checkbox") {
-        background-color: $mdc-checkbox-disabled-color-dark;
-      }
-      // stylelint-enable selector-max-compound-selectors, selector-max-specificity
-    }
+  @include mdc-checkbox-background-filled-enabled {
+    @include mdc-checkbox-stroke-color(secondary);
+    @include mdc-checkbox-container-color(secondary);
   }
 }
 
+// Needed to disable hover effects on CSS-only (non-JS) checkboxes
 .mdc-checkbox--disabled {
```

Comment:
how is `mdc-checkbox--disabled` different from `mdc-checkbox-background-filled-disabled` and `mdc-checkbox-background-unfilled-disabled`?

Reply:
`mdc-checkbox--disabled` is a modifier class for CSS-only buttons that prevents them from displaying a ripple or pointer cursor on hover.

`mdc-checkbox-unfilled-background-selector-disabled_` and friends are convenience mixins that wrap a block of Sass in the big ugly CSS selectors needed to target the 4 different state combinations. This will also hopefully help us keep the relatively complex selectors in sync by putting them all in one place.